src/GitHub.App/Services/GitHubContextService.cs

```diff
@@ -305,6 +307,55 @@ public bool TryOpenFile(string repositoryDir, GitHubContext context)
             }
         }
 
+        /// <inheritdoc/>
+        public string FindObjectishForTFSTempFile(string tempFile)
+        {
+            var match = tempFileObjectishRegex.Match(tempFile);
+            if (match.Success)
+            {
+                return match.Groups["objectish"].Value;
+            }
+
+            return null;
+        }
+
+        /// <inheritdoc/>
+        public (string commitSha, string blobPath) ResolveBlobFromHistory(string repositoryDir, string objectish)
+        {
+            using (var repo = gitService.GetRepository(repositoryDir))
+            {
+                var blob = repo.Lookup<Blob>(objectish);
+                if (blob == null)
+                {
+                    return (null, null);
+                }
+
+                foreach (var commit in repo.Commits)
+                {
+                    var trees = new Stack<Tree>();
+                    trees.Push(commit.Tree);
```

Comment:
I now know what a Git blob sha is, I've yet to wrap my head around a Commit Tree.

Reply:
I have to read the tests more